jetson/filter/src/__main__.py

```diff
@@ -164,6 +184,21 @@ def _driveVelDataCallback(self, channel, msg):
         drive_vel_data = DriveVelData.decode(msg)
         self.encoder_velocities[drive_vel_data.axis] = drive_vel_data.vel_percent
 
+    def _driveVelCmdCallback(self, channel, msg):
+        drive_vel_cmd = DriveVelCmd.decode(msg)
+        self.drive_v_left = drive_vel_cmd.left
+        self.drive_v_right = drive_vel_cmd.right
+
+    def _targetBearingCallback(self, channel, msg):
+        targetBearing = TargetBearing.decode(msg)
+        self.prev_target_bearing = self.target_bearing
+        self.target_bearing = targetBearing.target_bearing
+
+        target_diff = self.target_bearing - self.prev_target_bearing
+        self.target_bearing_rate = target_diff / \
+            (time.time() - self.last_target_bearing_fresh)
+        self.last_target_bearing_fresh = time.time()
```

Comment:
im worried about the state keeping being part of loc code instead of nav telling you the last requested bearing plus last requested time.

Reply:
yeah ig there might be a problem with time inaccuracy due to network latency and LCM packet loss. again, I'm not sure if we're even gonna need these metrics for the correction, do you think its worth adding this to nav before we even test it?

Reply:
i think it'll be easier to assess after we do the long time test saturday so probably fine not to add it in before, your call tho